ios/browser/api/ads/brave_ads.mm

```diff
@@ -473,18 +482,29 @@ - (void)removeAllHistory:(void (^)(BOOL))completion {
 
 #pragma mark - Observers
 
+- (void)addObserver:(ads::AdsClientObserver*)observer {
+  ads->AddObserver(observer);
+}
+
+- (void)removeObserver:(ads::AdsClientObserver*)observer {
+  ads->RemoveObserver(observer);
+}
+
+- (void)notifyDidInitializeAds {
+}
```

Comment:
1. These aren't used anywhere?
2. These aren't in the headers
3. We can't use it directly like this, we would need a wrapper around `ads::AdsClientObserver` in order to expose it in the headers/to brave-ios

Reply:
WIP. I did not mean to open the PR for review :-)